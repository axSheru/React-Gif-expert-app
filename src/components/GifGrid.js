import React from 'react'

export const GifGrid = ({ category }) => {

    const getGifs = async() => {
        const url = 'https://api.giphy.com/v1/gifs/search?q=The+mandalorian&limit=10&api_key=u1775mXvH3OtSHfVaM2EIIfeO4XCxjp9';
        const resp = await fetch( url );
        const { data } = await resp.json();

        const gifs = data.map( img => {
            return {
                id: img.id,
                title: img.title,
                url: img.images.downsized_medium.url,
            };
        });

        console.log(gifs);
    };

    getGifs();

    return (
        <div>
            <h3>{ category }</h3>
        </div>
    )
}
