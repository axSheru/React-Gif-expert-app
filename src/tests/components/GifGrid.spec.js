import { shallow } from "enzyme";
import { GifGrid } from "../../components/GifGrid";

describe('Pruebas en el componente GifGrid.', () => {

    let wrapper;
    const category = 'Netflix';

    beforeEach(() => {
        wrapper = shallow( <GifGrid category={ category } /> );
    });

    test('debe de hacer match con el snapshot.', () => {

        expect( wrapper ).toMatchSnapshot();

    });

});