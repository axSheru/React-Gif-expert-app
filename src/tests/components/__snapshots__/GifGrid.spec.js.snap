// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en el componente GifGrid. debe de hacer match con el snapshot. 1`] = `
<Fragment>
  <h3
    className="animate__animated animate__fadeIn"
  >
    Netflix
  </h3>
  <p
    className="animate__animated animate__flash"
  >
    Loading
  </p>
  <div
    className="card-grid"
  />
</Fragment>
`;
